parlai/chat_service/services/websocket/agents.py

```diff
@@ -39,38 +39,11 @@ def observe(self, act):
         """
         logging.info(f"Sending new message: {act}")
         quick_replies = act.get('quick_replies', None)
-        attachment_msg = self._get_attachment_msg(act)
-        if attachment_msg is not None:
-            self.manager.observe_payload(self.id, attachment_msg, quick_replies)
+        if act.get('payload', None):
+            self.manager.observe_payload(self.id, act['payload'], quick_replies)
         else:
             self.manager.observe_message(self.id, act['text'], quick_replies)
 
-    def _get_attachment_msg(self, act):
```

Comment:
what's the rationale for removing this?

Reply:
Removed this because rather than using the `attachment` key with either a `path` or `data` argument, we now accept a `payload` key with `data` and expect that the caller has loaded a base 64 encoded image in `data`.
Basically, shifting the responsibility so that the caller has to perform this logic which makes the payload more consistent with Messenger.
I just updated the documentation for `observe` to reflect the change.